media.go

```diff
@@ -964,6 +1000,112 @@ func (media *FeedMedia) Next(params ...interface{}) bool {
 	return false
 }
 
+// MediaItem defines a item media for the
+// SavedMedia struct
+type MediaItem struct {
+	Media Item `json:"media"`
+}
+
+// SavedMedia stores the information about media being saved before in my account.
+type SavedMedia struct {
+	inst     *Instagram
+	endpoint string
+
+	err error
+
+	Items []MediaItem `json:"items"`
+
+	NumResults          int    `json:"num_results"`
+	MoreAvailable       bool   `json:"more_available"`
+	AutoLoadMoreEnabled bool   `json:"auto_load_more_enabled"`
+	Status              string `json:"status"`
+
+	NextID interface{} `json:"next_max_id"`
+}
+
+// Next allows pagination
+func (media *SavedMedia) Next(params ...interface{}) bool {
+	// Inital error check
+	// if last pagination had errors
+	if media.err != nil {
+		return false
+	}
+
+	insta := media.inst
+	endpoint := media.endpoint
+	next := media.ID()
+
+	opts := &reqOptions{
+		Endpoint: endpoint,
+		Query: map[string]string{
+			"max_id": next,
+		},
+	}
+
+	body, err := insta.sendRequest(opts)
+	if err != nil {
+		media.err = err
+		return false
+	}
+
+	m := SavedMedia{}
+
+	if err := json.Unmarshal(body, &m); err != nil {
+		media.err = err
+		return false
+	}
+
+	*media = m
+
+	media.inst = insta
+	media.endpoint = endpoint
+	media.err = nil
+
+	if m.NextID == 0 || !m.MoreAvailable {
+		media.err = ErrNoMore
+	}
+
+	media.setValues()
+
+	return true
+}
+
+// Error returns the SavedMedia error
+func (media *SavedMedia) Error() error {
+	return media.err
+}
+
+// ID returns the SavedMedia next id
+func (media *SavedMedia) ID() string {
+	switch id := media.NextID.(type) {
+	case int64:
+		return strconv.FormatInt(id, 10)
+	case string:
+		return id
+	}
+	return ""
+}
+
+// Delete method TODO
+//
+// I think this method should use the
+// Unsave method, instead of the Delete.
+func (media *SavedMedia) Delete() error {
```

Comment:
Note that this method is not implemented. I don't really know what behavior is expected from this call. I think what makes sense is to unsave the items, as it's the equivalent of a _delete_ in the sense of saved items.

Reply:
So why did you wrote this function?

Reply:
Basically because it's part of the `Media` interface. For the moment it could return a not implemented error, for example. 
In the other hand, if you agree that the behavior should be unsave all the media items, I can implement it.